src/node_file.cc

```diff
@@ -255,6 +271,128 @@ void FileHandle::Close(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void FileHandle::ReleaseFD(const FunctionCallbackInfo<Value>& args) {
+  FileHandle* fd;
+  ASSIGN_OR_RETURN_UNWRAP(&fd, args.Holder());
+  // Just act as if this FileHandle has been closed.
+  fd->AfterClose();
+}
+
+
+void FileHandle::AfterClose() {
+  closing_ = false;
+  closed_ = true;
+  if (reading_ && !persistent().IsEmpty())
+    EmitRead(UV_EOF);
+}
+
+
+int FileHandle::ReadStart() {
+  if (!IsAlive() || IsClosing())
+    return UV_EOF;
+  reading_ = true;
+
+  if (current_read_ != nullptr)
+    return 0;
+
+  if (read_length_ == 0) {
+    EmitRead(UV_EOF);
+    return 0;
+  }
+
+  Local<Object> wrap_obj = env()->filehandlereadwrap_constructor()
+      ->NewInstance(env()->context()).ToLocalChecked();
+  {
+    AsyncHooks::DefaultTriggerAsyncIdScope trigger_scope(
+        env(), get_async_id());
+    current_read_ = new FileHandleReadWrap(this, wrap_obj);
+  }
+  int64_t recommended_read = 65536;
+  if (read_length_ >= 0 && read_length_ <= recommended_read)
+    recommended_read = read_length_;
+
+  current_read_->buffer = EmitAlloc(recommended_read);
+  current_read_->Dispatched();
+
+  uv_fs_read(env()->event_loop(),
+             current_read_->req(),
+             fd_,
+             &current_read_->buffer,
+             1,
+             read_offset_,
+             [](uv_fs_t* req) {
```

Comment:
Is creating a function here ok from a performance point of view? This is not the typical style I've used when working with libuv.

Reply:
@mcollina This doesn’t create a new function in C++.

> This is not the typical style I've used when working with libuv.

I think that’s just because a lot of our code that deals with libuv was written before we started using C++11.